Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyFragment.kt

```diff
@@ -62,79 +73,151 @@ class SurveyFragment : Fragment() {
-                        // content transform.
-                        when (targetState) {
-                            is SurveyState.Questions -> SurveyQuestionsScreen(
-                                questions = targetState,
-                                shouldAskPermissions = viewModel.askForPermissions,
-                                onAction = { id, action -> handleSurveyAction(id, action) },
-                                onDoNotAskForPermissions = { viewModel.doNotAskForPermissions() },
-                                onDonePressed = { viewModel.computeResult(targetState) },
-                                onBackPressed = {
-                                    activity?.onBackPressedDispatcher?.onBackPressed()
+
+                        SurveyQuestionsScreen(
+                            surveyScreenData = surveyScreenData,
+                            isNextEnabled = isNextEnabled,
+                            onClosePressed = {
+                                shouldInterceptBackPresses = false
+                                activity?.onBackPressedDispatcher?.onBackPressed()
+                            },
+                            onPreviousPressed = { viewModel.onPreviousPressed() },
+                            onNextPressed = { viewModel.onNextPressed() },
+                            onDonePressed = { viewModel.onDonePressed() }
+                        ) { paddingValues ->
+
+                            val modifier = Modifier.padding(paddingValues)
+
+                            AnimatedContent(
+                                targetState = surveyScreenData,
+                                transitionSpec = {
+                                    val animationSpec: TweenSpec<IntOffset> =
+                                        tween(CONTENT_ANIMATION_DURATION)
+                                    val direction = getTransitionDirection(
+                                        initialIndex = initialState.questionIndex,
+                                        targetIndex = targetState.questionIndex,
+                                    )
+                                    slideIntoContainer(
+                                        towards = direction,
+                                        animationSpec = animationSpec,
+                                    ) with slideOutOfContainer(
+                                        towards = direction,
+                                        animationSpec = animationSpec
+                                    )
                                 }
-                            )
-                            is SurveyState.Result -> SurveyResultScreen(
-                                result = targetState,
-                                onDonePressed = {
-                                    activity?.onBackPressedDispatcher?.onBackPressed()
+                            ) { targetState ->
+
+                                when (targetState.surveyQuestion) {
```

Comment:
The level of nesting here is quite extreme and makes it difficult to follow along. Would it be possible to extract a part of this method?